CMakeLists.txt

```diff
@@ -1200,10 +1198,13 @@ else()
     ${ZEPHYR_INCLUDE_DIRS}
   )
 
-  add_executable(       ${KERNEL_ELF} misc/empty_file.c ${GKSF})
-  target_link_libraries(${KERNEL_ELF} ${GKOF} ${TOPT} ${PROJECT_BINARY_DIR}/linker_pass_final.cmd ${zephyr_lnk} ${CODE_RELOCATION_DEP})
-  set_property(TARGET   ${KERNEL_ELF} PROPERTY LINK_DEPENDS ${PROJECT_BINARY_DIR}/linker_pass_final.cmd)
-  add_dependencies(     ${KERNEL_ELF} ${PRIV_STACK_DEP} ${LINKER_PASS_FINAL_SCRIPT_TARGET})
+  add_executable(       ${ZEPHYR_FINAL_EXECUTABLE} misc/empty_file.c ${GKSF})
+  target_link_libraries(${ZEPHYR_FINAL_EXECUTABLE} ${GKOF} ${TOPT} ${PROJECT_BINARY_DIR}/linker_pass_final.cmd ${zephyr_lnk} ${CODE_RELOCATION_DEP})
+  set_property(TARGET   ${ZEPHYR_FINAL_EXECUTABLE} PROPERTY LINK_DEPENDS ${PROJECT_BINARY_DIR}/linker_pass_final.cmd)
+  add_dependencies(     ${ZEPHYR_FINAL_EXECUTABLE} ${PRIV_STACK_DEP} ${LINKER_PASS_FINAL_SCRIPT_TARGET})
+
+  # Use the pass2 elf as the final elf
+  set(logical_target_for_zephyr_elf ${ZEPHYR_FINAL_EXECUTABLE})
```

Comment:
Either I'm totally off the mark, or these two lines are incredibly confusing because:

-  "pass2" in the comment == "FINAL" in the code, while
 - "final" in the comment == "logical" in the code.

Please correct or confirm.

Reply:
@marc-hb Correct, yes confusing. Initially I tried to rename the ZEPHYR_FINAL_EXECUTABLE to  ZEPHYR_EXECUTABLE_PASS2 (or similar), and ZEPHYR_PREBUILT_EXECUTABLE to ZEPHYR_EXECUTABLE_PASS1. But it was not accepted.

The logical target is the real final (in the words meaning) elf.
The confusion arises because 2nd isn't always the final, and the prebuilt 1st is the final.